node/src/components/consensus/protocols/highway.rs

```diff
@@ -710,4 +710,29 @@ where
     fn instance_id(&self) -> &C::InstanceId {
         self.highway.instance_id()
     }
+
+    fn recreate_timers(&self) -> Vec<ProtocolOutcome<I, C>> {
+        let mut outcomes = vec![ProtocolOutcome::ScheduleTimer(
+            Timestamp::now(),
+            TIMER_ID_PURGE_VERTICES,
+        )];
+
+        if self.highway.is_active() {
+            outcomes.push(ProtocolOutcome::ScheduleTimer(
+                Timestamp::now(),
+                TIMER_ID_ACTIVE_VALIDATOR,
+            ));
+        }
```

Comment:
I'm wondering whether this is necessary as well: if we didn't activate validators when `finished_joining = false` (which makes sense since they can't and shouldn't produce consensus messages while still joining) then once we are finished syncing we can activate the current era validator and return its effects.

Reply:
Maybe not, but still - the idea is to recreate everything that might have been there when this method is called, so I think it's better not to make that assumption.

Also, I think it's not that obvious - because if you're right and neither the active validator timer nor the synchronizer timers have to be recreated, that only leaves the purging timer which can be lost after the transition, and that shouldn't be breaking consensus, just causing it not to purge stuff, right? And consensus breaks if we don't recreate timers.

Reply:
I mean we do have to recreate the active validator timer and synchronizer timers but they can be "recreated" as a reaction to `FinishedJoining` event.

Reply:
They probably could, but I think that would conflate two very different things (see our discussion about `finished_joining` and `is_initialized` above).

Reply:
I simply think that this whole process can be simplified even further, without sacrificing correctness or readability.